interpreter/valid/valid.ml

```diff
@@ -598,9 +595,8 @@ let rec check_instr (c : context) (e : instr) (s : infer_result_type) : infer_in
     (ts1 @ [NumT I32T]) -->... [], []
 
   | ContNew x ->
-    let ct = cont_type c x in
-    let ft = func_type_of_cont_type c ct x.at in
-    [RefT (Null, heap_type_of_str_type c (DefFuncT ft))] -->
+    let (ContT ft) = cont_type c x in
+    [RefT (Null, ft)] -->
     [RefT (NoNull, DefHT (type_ c x))], []
```

Comment:
No need to look up the type again, in other places we also keep the syntactic type.
```suggestion
    [RefT (Null, ft)] --> [RefT (NoNull, VarHT (StatX x))], []
```